cpu/cortexm_common/include/cpu.h

```diff
@@ -189,6 +189,44 @@ static inline void cortexm_isr_end(void)
     }
 }
 
+/**
+ * @brief   Jumps to another image in flash
+ *
+ * This function is supposed to be called by a bootloader application.
+ *
+ * @param[in]   image_address   address in flash of other image
+ */
+static inline void cpu_jump_to_image(uint32_t image_address)
```

Comment:
The argument is `image_address` so should be declared as an address type. As there is currently no reason to limit to `32bit` or `cortexm_common` in the `cpu_jump_to_image` I would declare it as `void *`.

Reply:
> As there is currently no reason to limit to 32bit or cortexm_common in the cpu_jump_to_image I would declare it as void *.

Yes there is: it doesn't (and would never) exist a cortex-m cpu which is not 32bit. Moreover, only certain cortex-m (already ifdef'd) support the relocation of the vector table (VTOR). Thus this code is not meant to be executed by any other architecture.

For more information: http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0552a/Ciheijba.html however they measure the offset in words instead of bytes.

Reply:
Seems like this function should be called `jump_to_address`? Since it's being passed an address, not an image number